### PR TITLE
[RFC] doc: Remove os_unix.txt

### PR DESCRIPTION
**Pull request by ghost**
This is done in a bunch of commits in order to ease review; see the individual commit messages for more info. I can squash everything before merging if wanted.

### CODE REVIEW

**Comment by fmoralesc** [comment]
LFGTM. :+1:

**Comment by Pyrohh** [comment]
Thanks, I'll merge this in a day or two if there are no objections.

File: runtime/doc/os_unix.txt [hunk 1]
```diff
@@ -1,57 +0,0 @@
...
-For compiling Vim on Unix see "INSTALL" and "Makefile" in the src directory.
-
-The default help file name is "/usr/local/lib/vim/help.txt" The files
-"$HOME/.vimrc" and "$HOME/.exrc" are used for configuration.  Additionally
-"/usr/local/etc/vimrc" is used first.  If "/usr/local/share" exists it is used
-instead of "/usr/local/lib".
-
-Temporary files (for filtering) are put in "/tmp".  If you want to place them
-somewhere else, set the environment variable $TMPDIR to the directory you
-prefer.
```
Comment (justinmk) [inline runtime/doc/os_unix.txt]: this is documented at `:h tempfile`

**Comment by justinmk** [comment]
LGTM after squash

File: runtime/doc/os_unix.txt (hunk 1 above)
Comment (Pyrohh) [inline runtime/doc/os_unix.txt]: Indeed. I prefer it that way, being centralized and all.

**Comment by Pyrohh** [comment]
Thanks for the reviews :)

**Pull request merged**